core/lib/multivm/src/versions/vm_latest/tracers/dispatcher.rs

```diff
@@ -0,0 +1,114 @@
+use crate::interface::dyn_tracers::vm_1_3_3::DynTracer;
+use crate::interface::tracer::{TracerExecutionStatus, VmExecutionStopReason};
+use crate::vm_latest::{BootloaderState, HistoryMode, SimpleMemory, VmTracer, ZkSyncVmState};
+use std::cell::RefCell;
+use std::rc::Rc;
+use zk_evm_1_3_3::tracing::{
+    AfterDecodingData, AfterExecutionData, BeforeExecutionData, VmLocalStateData,
+};
+use zksync_state::{StoragePtr, WriteStorage};
+
+pub type TracerPointer<S, H> = Rc<RefCell<dyn VmTracer<S, H>>>;
+
+impl<S: WriteStorage, H: HistoryMode> From<TracerPointer<S, H>> for TracerDispatcher<S, H> {
+    fn from(value: TracerPointer<S, H>) -> Self {
+        Self {
+            tracers: vec![value],
+        }
+    }
+}
+
+impl<S: WriteStorage, H: HistoryMode> From<Vec<TracerPointer<S, H>>> for TracerDispatcher<S, H> {
+    fn from(value: Vec<TracerPointer<S, H>>) -> Self {
+        Self { tracers: value }
+    }
+}
+
+pub struct TracerDispatcher<S: WriteStorage, H: HistoryMode> {
+    pub(crate) tracers: Vec<TracerPointer<S, H>>,
+}
```

Comment:
instead of `pub(crate)` probably better `AsRef<[TraicerPointer]>` and `AsRef<Vec<TracerPointer>>`